net/netcheck/netcheck.go

```diff
@@ -980,6 +1015,33 @@ func (c *Client) finishAndStoreReport(rs *reportState, dm *tailcfg.DERPMap) *Rep
 	return report
 }
 
+const generate204URL = "http://controlplane.tailscale.com/generate_204"
+
+var noRedirectClient = &http.Client{
+	// No redirects allowed
+	CheckRedirect: func(req *http.Request, via []*http.Request) error {
+		return http.ErrUseLastResponse
+	},
+
+	// Remaining fields are the same as the default client.
+	Transport: http.DefaultClient.Transport,
+	Jar:       http.DefaultClient.Jar,
+	Timeout:   http.DefaultClient.Timeout,
+}
+
+func (c *Client) checkCaptivePortal(ctx context.Context) (bool, error) {
+	ctx, cancel := context.WithTimeout(ctx, captivePortalTimeout)
+	defer cancel()
+
+	req, _ := http.NewRequestWithContext(ctx, "GET", generate204URL, nil)
+	r, err := noRedirectClient.Do(req)
+	if err != nil {
+		return false, nil
```

Comment:
Return err? Or drop the error return

Reply:
Oops; was supposed to return that. I want to distinguish between "error making a request" and "positive response that isn't the status code we expect".